### PR TITLE
Method lookup table

### PR DESCRIPTION
**Pull request by maximpertsov**
This branch is rebased from `fix-start-ok`, so make that one gets merged in first.

With this change Class/Method enum values are no longer significant. This unblocks us from adding Methods from the AMQP spec that have the same Method IDs.

### CODE REVIEW

File: rmqsrc/spec.nim [hunk 1]
```diff
@@ -33,3 +36,29 @@ const
 
   FRAME_MAX_SIZE* = 131072
   BODY_MAX_LENGTH* = FRAME_MAX_SIZE - FRAME_HEADER_SIZE - FRAME_END_SIZE
+
+let
```
Comment (maximpertsov) [inline rmqsrc/spec.nim:40]: We can't define `METHODS` and its reverse lookup counterpart `METHOD_VALS` as consts if we want to define one from the other. Kind of a bummer...

File: rmqsrc/decode.nim [hunk 2]
```diff
@@ -200,11 +200,17 @@ proc decodeConnectionStart(data: Stream): Method =
 
 proc decodeMethod(data: Stream): Method =
   let
-    class = data.readClassId.Class
-    methodId = data.readMethodId.MethodId
+    classNum = data.readClassId
+    methodNum = data.readMethodId
+
+  let
+    pair = toMethod(classNum, methodNum)
```
Comment (subsetpark) [inline rmqsrc/decode.nim]: This can be expressed as `let (class, methodId) = ...`

File: rmqsrc/spec.nim [hunk 3]
```diff
@@ -33,3 +36,29 @@ const
...
+    (10.uint16, 40.uint16): (cConnection, mOpen),        # open connection to virtual host
+    (10.uint16, 41.uint16): (cConnection, mOpenOk),      # signal that connection is ready
+    (10.uint16, 50.uint16): (cConnection, mClose),       # request a connection close
+    (10.uint16, 51.uint16): (cConnection, mCloseOk)      # confirm a connection close
+  }.toTable
+
+  METHOD_VALS = toSeq(METHODS.pairs).mapIt((it[1], it[0])).toTable
+
+proc toMethod*(cNum: uint16, mNum: uint16): (Class, MethodId)  =
+  if not METHODS.hasKey((cNum, mNum)):
```
Comment (subsetpark) [inline rmqsrc/spec.nim]: `in` works here like in Python

File: rmqsrc/decode.nim (hunk 2 above)
Comment (maximpertsov) [reply rmqsrc/decode.nim]: Ah good to know... I tried to do this without the parens before and it didn't work

**Pull request merged**